platform/linux-generic/odp_timer.c

```diff
@@ -76,7 +76,7 @@ static _odp_atomic_flag_t locks[NUM_LOCKS]; /* Multiple locks per cache line! */
 #endif
 
 /* Max timer resolution in nanoseconds */
-static uint64_t highest_res_ns;
+static uint64_t highest_res_ns = 500;
```

Comment:
This doesn't do what you say you want since the `timer_res_init()` routine overwrites this. Sounds like what you want is to introduce a `max_res_ns` limit (setting it to 500) and then use that to override the computed `highest_res_ns` if `timer_res_init()` computes a lower value. So timer pools may have a higher `highest_res_ns` but they cannot have one less than `max_res_ns`.

Reply:
When timer uses posix timer, this value is measured and I'm OK what ever the measurement results (won't be nano- but milliseconds anyway). When timer won't use posix but rely on polls from scheduler, this value is not measured and was left to zero. This fix is for that mode.

Reply:
OK, I missed that subtlety in the code. A comment would be useful to make it clear what the intent is here.